tests/integration/tls_tests/test_tls.py

```diff
@@ -255,7 +260,7 @@ async def test_set_tls_key(ops_test: OpsTest) -> None:
         new_internal_cert_time = await time_file_created(ops_test, unit.name, INTERNAL_CERT_PATH)
         new_mongod_service_time = await time_process_started(ops_test, unit.name, DB_SERVICE)
 
-        check_certs_correctly_distributed(ops_test, unit)
+        await check_certs_correctly_distributed(ops_test, unit)
```

Comment:
@Mehdi-Bendriss does this work? IIRC there were extensive changes needed for this to work on VM

Reply:
I think It should as there is no sharding involved only internal vs external TLS resources

Reply:
When we co-worked on the [fix TLS Pr for VM](https://github.com/canonical/mongodb-operator/pull/395/files#diff-9f99a813697ed7cb55d5b5fa50b90b7f37c8c532c2c9fbac05b4b87ab5a11e36) the extensive changes for tests to pass were also applied to the non-sharding tests.

We can separate it out into a separate PR and I can fix it there, but to keep the commit history on this repo good, I would kindly request that you change this PR name to `[DPE-4189] update charm components - move to DP `